build/ci_utils/src/actions/artifacts.rs

```diff
@@ -185,14 +183,15 @@ pub fn single_file_provider(
 pub fn single_dir_provider(path: &Path) -> Result<impl Stream<Item = FileToUpload> + 'static> {
     // TODO not optimal, could discover files at the same time as handling them.
     let parent_path = path.try_parent()?;
-    let files = walkdir::WalkDir::new(path)
-        .into_iter()
-        .try_collect_vec()?
+    let files: Vec<FileToUpload> = walkdir::WalkDir::new(path)
         .into_iter()
-        .filter(|entry| !entry.file_type().is_dir())
-        .map(|entry| FileToUpload::new_relative(parent_path, entry.path()))
-        .try_collect_vec()?;
-
+        .filter(|res| match res {
+            Ok(entry) => !entry.file_type().is_dir(),
+            // ignore "not found" errors, just don't include the file in the final stream.
+            Err(e) => e.io_error().map_or(true, |e| e.kind() != std::io::ErrorKind::NotFound),
```

Comment:
This case suggests a race condition; should we log a warning?

Reply:
It's not a race condition. The file can simply not exist if you ask for an upload of a missing directory.

Reply:
Recognizing that single-error case specifically rather than filtering would avoid potentially hiding real errors.

Reply:
It is recognizing that single-error case. That's the only error that's filtered out. All other errors will be passed down the iterator, eventually bubbling up throught `try_collect`.

Reply:
It's filtering out arbitrarily many errors.

Reply:
Okay, I see what you mean. Yes, if there are multiple "not found" errors, those are all filtered out. I don't see how else to do that though, given the `walkdir` API. I could additionaly check if the error is at depth 0 (i.e. definitely tied to the root directory). The only other option I can think of is to simply discard all artifact upload errors, as we used to do before (the result from promise was simply thrown out).